src/main/java/teammates/storage/sqlentity/Instructor.java

```diff
@@ -0,0 +1,237 @@
+package teammates.storage.sqlentity;
+
+import java.time.Instant;
+import java.util.Map;
+import java.util.Objects;
+
+import org.hibernate.annotations.CreationTimestamp;
+import org.hibernate.annotations.UpdateTimestamp;
+
+import com.googlecode.objectify.annotation.Entity;
+import com.googlecode.objectify.annotation.Id;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.EnumType;
+import jakarta.persistence.Enumerated;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.OneToOne;
+import jakarta.persistence.PrimaryKeyJoinColumn;
+import jakarta.persistence.Table;
+import teammates.ui.output.InstructorPermissionRole;
+
+/**
+ * Represents an Instructor entity.
+ */
+@Entity
+@Table(name = "Instructors")
+// Might have to change this. Should follow User's PK. Same for Student
+@PrimaryKeyJoinColumn(name = "userId")
+public class Instructor { // TODO: extends User
```

Comment:
Let's get the User class up as well before we continue.

Model this as you would using the OOP paradigm. Specifically,

```
public abstract class User {
  String id;
  ...
}
```
```
public class Instructor extends User {
  String registrationKey;
  boolean isDisplayedToStudents;
  ...
}
```

Notice we do not need any of the fields in the parent class, these are inherited as per OOP rules.

Once you are done, add the relevant annotations. Since you have chosen Joined Table here,

```
@Entity
@Inheritance(strategy = InheritanceType.JOINED)
public abstract class User {
...
}
```

For instructor, just add
```
@Entity
public class Instructor {
...
}
```

Let hibernate take care of the mapping to tables for us.